### PR TITLE
Module 4 - Subcategories

### PR DESCRIPTION
**Pull request by sinpai**
* Made categories polymorphic
* Added task in rake to convert them in new type
* Added subcategories on a category history page

Screens - http://take.ms/Yy3k6
Video - http://take.ms/yB26o

### CODE REVIEW

**Review by AleksSenkou — CHANGES_REQUESTED** [review]
Looks good to merge,
But I can't test those changes, see [trello card](https://trello.com/c/CSAj7sib/24-module-4-%D0%BF%D0%B5%D1%80%D0%B5%D0%B2%D0%BE%D0%B4-%D1%81%D1%80%D0%B5%D0%B4%D1%81%D1%82%D0%B2-%D0%B2-%D0%BA%D0%B0%D1%82%D0%B5%D0%B3%D0%BE%D1%80%D0%B8%D1%8E-%D1%81-%D0%BE%D0%B1%D1%89%D0%B5%D0%B3%D0%BE-%D0%B1%D0%B0%D0%BB%D0%B0%D0%BD%D1%81%D0%B0)

**Comment by AleksSenkou** [comment]
@sinpai I've added new comments with a bug to [trello card](https://trello.com/c/7oYUSpU5/27-module-4-%D0%BF%D0%BE%D0%B4%D0%BA%D0%B0%D1%82%D0%B5%D0%B3%D0%BE%D1%80%D0%B8%D0%B8)

File: app/controllers/categories_controller.rb [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
+  before_action :find_category, only: %i[update destroy edit show]
+  before_action :set_parent, only: %i[new create]
 
   def index
-    @categories = current_user.categories.paginate(page: params[:page], per_page: 10)
+    @categories = current_user.categories.top_category.paginate(page: params[:page], per_page: 10)
+  end
+
+  def show
+    @transactions = current_user.transactions.category_history_transactions(params[:id].to_i)
```
Comment (nickzhihovich) [inline app/controllers/categories_controller.rb:11]: @sinpai Q about `to_i` from previous PRs

File: config/locales/en.yml [hunk 2]
```diff
@@ -76,7 +76,7 @@ en:
       edit_category: 'Edit category'
     create:
       category_created: 'You successfully create category.'
-      category_not_created: 'Category not create'
+      category_not_created: 'Category cannot be created. Please enter more than 2 characters.'
```
Comment (nickzhihovich) [inline config/locales/en.yml]: @sinpai I think it's bad errors message. if I enter more than 40 characters I get an error "...enter more than 2 characters". Let's change it.

**Review by nickzhihovich — COMMENTED** [review]
suggestions

File: app/controllers/categories_controller.rb (hunk 1 above)
Comment (sinpai) [reply app/controllers/categories_controller.rb:11]: http://take.ms/nGVyy - it's a string. But I need integer in this case.

**Review by AleksSenkou — APPROVED** [review]
LGTM

**Pull request merged**